services/121-service/src/payments/dto/get-transaction-response.dto.ts

```diff
@@ -0,0 +1,41 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+import { TransactionStatusEnum } from '@121-service/src/payments/transactions/enums/transaction-status.enum';
+import { RegistrationStatusEnum } from '@121-service/src/registration/enum/registration-status.enum';
+
+export class GetTransactionResponseDto {
+  @ApiProperty({ example: 1, type: 'number' })
+  public id: number;
+  @ApiProperty({
+    example: '2023-09-28T08:00:10.363Z',
+    type: 'string',
+    format: 'date-time',
+  })
+  public created: Date;
+  @ApiProperty({
+    example: '2023-09-28T08:00:10.363Z',
+    type: 'string',
+    format: 'date-time',
+  })
+  public updated: Date;
+  @ApiProperty({ example: 1, type: 'number' })
+  public payment: number;
+  @ApiProperty({ example: TransactionStatusEnum.success, type: 'string' })
+  public status: TransactionStatusEnum;
+  @ApiProperty({ example: 25, type: 'number' })
+  public amount: number;
+  @ApiProperty({ example: 'Something went wrong', type: 'string' })
+  public errorMessage: string | null;
+  @ApiProperty({ example: 'ironBank' })
+  public programFinancialServiceProviderConfigurationName: string;
+  @ApiProperty({ example: 111, type: 'number' })
+  public registrationProgramId: number;
+  @ApiProperty({ example: 111, type: 'number' })
+  public registrationId: number;
+  @ApiProperty({ example: RegistrationStatusEnum.included, type: 'string' })
+  public registrationStatus: RegistrationStatusEnum;
+  @ApiProperty({ example: '12345', type: 'string' })
+  public referenceId: string;
```

Comment:
@diderikvw You suggested to name this to registrationReferenceId and I also think it makes sense. 

However naming registrationReferenceId made a lot of API and such fail. Because a lot of these use referenceId to match whether a transaction belongs to a certain registration. 

Renaming this is a bit of work for which I do not have time before my holidays and I don't want to hand you guys a half working branch, so therefore leaving it as referenceId for now.

Reply:
And example of where is goes wrong can be found in this function: test/helpers/program.helper.ts -> waitForPaymentTransactionsToComplete